cdap-data-fabric/src/test/java/io/cdap/cdap/data2/dataset2/lib/cube/AbstractCubeTest.java

```diff
@@ -398,6 +399,128 @@ public void testMetricDeletion() throws Exception {
+                          ImmutableMap.of("metric1", AggregationFunction.SUM),
+                          ImmutableMap.of("dim1", "tag1"), ImmutableList.of("dim2"),
+                          new MetricsAggregationOption(5, MetricsAggregationOption.AggregationOption.SUM),
+                          null);
+    result = new ArrayList<>(cube.query(query));
+    Assert.assertEquals(2, result.size());
+    // agg1 gets increment by 1 for 100 seconds, so sum will be 100/5=20, agg2 gets increment by 3 for 50 seconds, so
+    // sum will be 3*50/5=30
+    verifySumAggregation(result.get(0), "metric1", 5, 30, 10, 0);
+    verifySumAggregation(result.get(1), "metric1", 5, 20, 20, 0);
+
+    // test metric1 with count 9, this will create a remainder since 100/9=11, so there will be 10 data points returned
+    query = new CubeQuery(null, 0, 200, 1, Integer.MAX_VALUE,
+                          ImmutableMap.of("metric1", AggregationFunction.SUM),
+                          agg1Dims, Collections.emptyList(),
+                          new MetricsAggregationOption(9, MetricsAggregationOption.AggregationOption.SUM),
+                          null);
+    result = new ArrayList<>(cube.query(query));
+    Assert.assertEquals(1, result.size());
+    // the data points will be 10, with 9 of them has sum 11, and the remainder will have sum 1
+    verifySumAggregation(result.get(0), "metric1", 10, 11, 11, 1);
+  }
+
+  @Test
+  public void testMetricsAggregationOptionLatest() throws Exception {
+    Aggregation agg = new DefaultAggregation(ImmutableList.of("dim1", "dim2", "dim3"),
+                                             ImmutableList.of("dim1"));
+
+    int resolution = 1;
+    Cube cube = getCube("testAggOptionLatest", new int[] {resolution}, ImmutableMap.of("agg", agg));
+
+    Map<String, String> aggDims = new LinkedHashMap<>();
+    aggDims.put("dim1", "tag1");
+    aggDims.put("dim2", "tag2");
+    aggDims.put("dim3", "tag3");
+
+    // write 100 data points to agg
+    for (int i = 1; i <= 100; i++) {
+      writeGauge(cube, "metric1",  i,  i,  aggDims);
+    }
+
+    // query for latest, should have the latest value for each interval, 20, 40, 60, 80, 100
+    CubeQuery query = new CubeQuery(null, 0, 200, 1, Integer.MAX_VALUE,
+                                    ImmutableMap.of("metric1", AggregationFunction.SUM),
+                                    aggDims, Collections.emptyList(),
+                                    new MetricsAggregationOption(5,
+                                                                 MetricsAggregationOption.AggregationOption.LATEST),
+                                    null);
+    List<TimeSeries> result = new ArrayList<>(cube.query(query));
+    Assert.assertEquals(1, result.size());
+    List<TimeValue> timeValues = result.get(0).getTimeValues();
+    for (int i = 0; i < timeValues.size(); i++) {
+      Assert.assertEquals(20 * (i + 1), timeValues.get(i).getValue());
+    }
+  }
+
```

Comment:
you should test the behavior when 0, a negative number, or a number greater than the query limit is specified

Reply:
Added